src/ring_theory/ideal/local_ring.lean

```diff
@@ -406,6 +410,17 @@ is the residue field of `R`. -/
   map_mul' := λ e₁ e₂, map_equiv_trans e₂ e₁,
   map_one' := map_equiv_refl }
 
+section mul_semiring_action
+variables (G : Type*) [group G] [mul_semiring_action G R]
+
+/-- If `G` acts on `R` as a `mul_semiring_action`, then it also acts on `residue_field R`. -/
+noncomputable instance : mul_semiring_action G (local_ring.residue_field R) :=
+mul_semiring_action.comp_hom _ $ map_aut.comp (mul_semiring_action.to_ring_aut G R)
+
+lemma residue_smul (g : G) (r : R) : residue R (g • r) = g • residue R r := rfl
```

Comment:
I think this should be stated in reverse, and potentially be a simp lemma

Reply:
OK, thanks. I added simp in both lemmata

Reply:
I don't agree with flipping this; having it this way around makes it match `map_add residue`

Reply:
My thought was that this lemma unfolds the definition above. But I guess what you said makes more sense. Let's flip it back.